homeassistant/components/sensor/openexchangerates.py

```diff
@@ -0,0 +1,95 @@
+"""Support for openexchangerates.org exchange rates service."""
+from datetime import timedelta
+import logging
+import requests
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle
+from homeassistant.const import CONF_API_KEY
+
+_RESOURCE = 'https://openexchangerates.org/api/latest.json'
+_LOGGER = logging.getLogger(__name__)
+# Return cached results if last scan was less then this time ago.
+MIN_TIME_BETWEEN_UPDATES = timedelta(minutes=100)
+CONF_BASE = 'base'
+CONF_QUOTE = 'quote'
+CONF_NAME = 'name'
+DEFAULT_NAME = 'Exchange Rate Sensor'
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the Openexchangerates sensor."""
+    payload = config.get('payload', None)
+    rest = OpenexchangeratesData(
+        _RESOURCE,
+        config.get(CONF_API_KEY),
+        config.get(CONF_BASE, 'USD'),
+        config.get(CONF_QUOTE),
+        payload
+    )
+    rest.update()
+    add_devices([OpenexchangeratesSensor(rest,
```

Comment:
I added the status code check on lines 88 through 90 and the sensor is not added when the API key is not valid, but it does not exit gracefully. 

```
16-06-22 20:46:34 custom_components.sensor.openexchangerates: Check OpenExchangeRates API
16-06-22 20:46:34 homeassistant.components.sensor: Error while setting up platform openexchangerates
Traceback (most recent call last):
  File "/usr/local/lib/python3.4/dist-packages/homeassistant/helpers/entity_component.py", line 98, in _setup_platform
    discovery_info)
  File "/home/pi/.homeassistant/custom_components/sensor/openexchangerates.py", line 32, in setup_platform
    config.get(CONF_QUOTE))])
  File "/home/pi/.homeassistant/custom_components/sensor/openexchangerates.py", line 43, in __init__
    self.update()
  File "/home/pi/.homeassistant/custom_components/sensor/openexchangerates.py", line 64, in update
    self._state = round(value[str(self._quote)], 4)
TypeError: 'NoneType' object is not subscriptable
```

I should probably do some check around 29, but the `try` and `except` did not work. Any suggestions?